### DISCUSSION TITLE
Programatically turn on and off instance

### DISCUSSION THREAD

**Discussion by oli414** [mozilla/hubs#3643] [Hubs Cloud Questions] [answered]
Is it possible to programatically turn on and off the the Hubs Clouds instance?

We have a Hubs environment that has to be available on demand, we were thinking of having a button on a page to put the hubs clouds instance online, or offline when it isn't in use. Is it possible to do this?

**Comment by robinkwilson** [comment, marked as answer]
Yes it is. I'd recommend looking at the AWS CLI or AWS SDK for Python or Javascript to make a script that Updates the Cloudformation stack to change the instance from online mode to offline mode and vice versa. 

The tricky part will be securing your AWS credentials because you'd be giving the script and anyone who clicks the button a LOT of power over your AWS account. Possibly setting up an IAM role that can only do this action would help. 

Hope this helps! Good luck!
